packages/flame_3d/lib/src/components/light_component.dart

```diff
@@ -0,0 +1,38 @@
+import 'package:flame_3d/camera.dart';
+import 'package:flame_3d/components.dart';
+import 'package:flame_3d/game.dart';
+import 'package:flame_3d/resources.dart';
+
+/// A [Component3D] that represents a light source in the 3D world.
+class LightComponent extends Component3D {
+  LightComponent({
+    required this.source,
+    super.position,
+  });
+
+  LightComponent.spot({
+    Vector3? position,
+  }) : this(
+          source: SpotLight(),
+          position: position,
+        );
+
+  final LightSource source;
+
+  late final Light _light = Light(
+    transform: Transform3D(),
+    source: source,
+  );
+
+  Light get light {
+    return _light..transform.setFrom(transform);
+  }
+
+  @override
+  void onMount() {
+    assert(
+      parent is World3D,
+      'Lights must be added to the root of the World3D',
+    );
```

Comment:
Instead of an assert we should maybe just throw an exception? Asserts dissapear in release mode and children can be dynamically added/removed so you might never know why it fails in release mode

Reply:
this is the idiomatic way we do elsewhere, so I'd rather keep it consistent and change all at once later